src/azure-cli/azure/cli/command_modules/vm/_image_builder.py

```diff
@@ -370,16 +376,47 @@ def process_img_tmpl_output_add_namespace(cmd, namespace):
 
 # region Custom Commands
 
-def create_image_template(  # pylint: disable=too-many-locals
+def create_image_template(  # pylint: disable=too-many-locals, too-many-branches, too-many-statements
         cmd, client, resource_group_name, image_template_name, location=None,
         source_dict=None, scripts_list=None, destinations_lists=None, build_timeout=None, tags=None,
         source=None, scripts=None, checksum=None, managed_image_destinations=None,  # pylint: disable=unused-argument
-        shared_image_destinations=None, no_wait=False):  # pylint: disable=unused-argument, too-many-locals
+        shared_image_destinations=None, no_wait=False, image_template=None):  # pylint: disable=unused-argument, too-many-locals
     from azure.mgmt.imagebuilder.models import (ImageTemplate, ImageTemplateSharedImageVersionSource,
                                                 ImageTemplatePlatformImageSource, ImageTemplateIsoSource, ImageTemplateManagedImageSource,  # pylint: disable=line-too-long
                                                 ImageTemplateShellCustomizer, ImageTemplatePowerShellCustomizer,
                                                 ImageTemplateManagedImageDistributor, ImageTemplateSharedImageDistributor)  # pylint: disable=line-too-long
 
+    if image_template is not None:
+        if os.path.exists(image_template):
+            # Local file
+            with open(image_template) as f:
+                content = f.read()
+        else:
+            # It should be an URL
+            msg = '\nusage error: --image-template is not a correct local path or URL'
+            try:
+                r = requests.get(image_template)
+            except Exception:
+                raise CLIError(traceback.format_exc() + msg)
+            if r.status_code != 200:
+                raise CLIError(traceback.format_exc() + msg)
+            content = r.content
+
+        try:
+            obj = json.loads(content)
+        except json.JSONDecodeError:
+            raise CLIError(traceback.format_exc() +
+                           '\nusage error: Content of --image-template is not a valid JSON string')
+        content = {}
+        if 'properties' in obj:
+            content = obj['properties']
+        if 'location' in obj:
+            content['location'] = obj['location']
+        if 'tags' in obj:
+            content['tags'] = obj['tags']
+        return client.virtual_machine_image_templates.create_or_update(
```

Comment:
seems `content` is not used in rest call, if true, is it necessray to set it up, could directly checking schema of obj.

Reply:
content is use in `parameters=content`.

Reply:
sorry, my fault